src/main/kotlin/org/rust/ide/intentions/ExpandModule.kt

```diff
@@ -17,7 +20,7 @@ class ExpandModule : IntentionAction {
     override fun startInWriteAction()   = true
 
     override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?) =
-        file is RustFile && file.name != RustModules.MOD_RS
+        file is RustFile && (!(file.rustMod as RustModItem).ownsDirectory && !(file.name.equals(RustModules.LIB_RS) && file.containingDirectory?.name.equals("src")))
```

Comment:
If you use `ownsDirectory`, then you don't need to check file name or parent directory name. 

Also this `as` casts away nullability and is effectively just `!!`. You can use something like `file.rustMod?.ownsDirectory ?: false`